### ISSUE TITLE
Bug: Line breaks between lists and horizontal rules are removed when "Remove empty lines between list markers and checklists" is enabled

### ISSUE THREAD

**Issue by todd-pd** [platers/obsidian-linter#283]
## Describe the Bug

Line breaks between lists and horizontal rules are removed when "Remove empty lines between list markers and checklists" is enabled, if the same character is used for the list item and for the horizontal rule.

## How to Reproduce

Given I have `"Remove empty lines between list markers and checklists"` enabled, and I have Markdown like:

```markdown
- Some list item 1
- Some list item 2

---

Some text
```

When I trigger the linter by saving the file

Then the content is linted like this (notice the line break removed between the list and the rule):

```markdown
- Some list item 1
- Some list item 2
---

Some text
```

## Expected Behavior

Given the above example, I would have expected there to be no linting changes.

## Screenshots
N/A

## Device

- [x] Desktop
- [ ] Mobile

## Additional Context

This can be avoided by ensuring that the punctuation used for adjacent lists and rules is different (e.g. `***` and `-`, or `---` and `*`). But it forces you to change the way you write your Markdown, so it's not an ideal workaround.

**Comment by pjkaufman** [comment]
Thanks for reporting this issue, @todd-pd . I will see what is going on. It is likely a regex check gone awry.

**Comment by pjkaufman** [comment]
I will go ahead and ignore hr's/thematic breaks for this to fix the issue.

**Comment by pjkaufman** [comment]
@todd-pd , are you using the example at the start of the file? If so, it might have the yaml affecting it. Otherwise, I agree that it is the rule you mentioned.

**Comment by todd-pd** [comment]
@pjkaufman No, the example is slightly misleading in that way. There's more plain text above the first horizontal rule. No YAML.

**Comment by pjkaufman** [comment]
Gotcha. This should be fixed now on master. If the issue is still present there or in the next release, please feel free to let us know and reopen this issue.

**Comment by todd-pd** [comment]
Nice, thanks for the quick turnaround!

**Comment by pjkaufman** [comment]
No problem. It was a simple fix.
